Makefile

```diff
@@ -49,3 +54,43 @@ codegen-internal:
 	cp boilerplate.go.txt vendor/k8s.io/code-generator/hack/boilerplate.go.txt
 	$(CODEGEN_INTERNAL) deepcopy,conversion,client,lister,informer $(PKG)/pkg/generated $(PKG)/pkg/apis $(PKG)/pkg/apis "autoscaling:v1"
 
+
+deploy-local: DEPLOY_MODE := local
+deploy-local: deploy
+
+deploy-ci: DEPLOY_MODE := ci
+deploy-ci: deploy
```

Comment:
Instead of deploying "manually" in CI can you install via OLM/Operator framework by, for example (I'm not sure of all teh details), creating an `OperatorSource` and `Subscription` ?

Reply:
This manual deployment ensures that the operator works as expected ( without any OLM machinery involved ) which provides a quick way to flush out any regression bugs or permission issues with the kube manifests. 

I also have plan to add another goal `make e2e-olm` that will install the operator using OLM (`CatalogSource`, `Subscription` object). That's a little bit more challenging though but it will help us flush out any issues with the generated OLM manifests and identify issues with OLM integration.